### PR TITLE
iss1137 form documentation update

### PR DESCRIPTION
**Pull request by mmenestr**
This PR would fix issue #1137 

Hoping this one doesn't have any issues with it...

### CODE REVIEW

File: packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md [hunk 1]
```diff
@@ -5,254 +5,280 @@ hideTOC: true
...
+* Always provide a label for input components.
+* Labels should clearly state the required input.
+* **Do not** use a colon after the label name.
 
-If you have more than 5 options to present to the user, use a select list.
+### Required fields
+A required field should be indicated with an asterisk ( * ) to the right of the field label.
 
-**Radio buttons**
+![Example of field with required field indication](./img/required-field.png)
```
Comment (mcarrano) [inline packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md:118]: In this image, the asterisk seems very far away from the label compared with the implementation here: https://www.patternfly.org/v4/documentation/react/components/form Is this what is coming from the Sketch library?

File: packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md [hunk 2]
```diff
@@ -5,254 +5,280 @@ hideTOC: true
...
+Validation on submission is performed when the user attempts to submit a form. In this scenario, the user completes the form in its entirety and the page is reloaded with any errors detected. 
 
-  **DON'T**
+For validation on submission, use an [inline alert](design-guidelines/usage-and-behavior/alerts-and-notifications#inline-alerts) along with field level errors wherever possible to make it simple for the user to quickly identify areas that need to be fixed.
+Notifications should describe the error and help users solve the problem.
 
-  * Don’t use placeholder text to communicate complex and lengthy requirements like password requirements. Instead, use an infotip.
-  * Don’t provide placeholder text when it isn't necessary.
-  * **Never** use placeholder text as a replacement for field labels.
+![Example of errors after validation on submission, including an inline alert at the top, and field level errors](./img/error-validation.png)
```
Comment (mcarrano) [inline packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md:204]: Same issue here with required field asterisks

File: packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md (hunk 1 above)
Comment (mmenestr) [reply packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md:118]: Yes, took everything straight out of the sketch library!

Comment (mcarrano) [reply packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md:118]: Perhaps this is also a fix we need to make in Sketch then. @gdoyle1 see my comment. Has this always been this way? Sketch and the code are different then. Which do we think is correct?

**Comment by mcarrano** [comment]
@mmenestr Something odd is happening here. I see that you updated the images but that's not what I see when I preview the page. It may be a problem with this code: img src="required-field.png  Shouldn't the src be img/required-field.png?

**Comment by mmenestr** [comment]
@mcarrano just updated to what you said - let me know if that does the trick. Your change actually allowed the images to show up in the "preview" of the code as well !

File: packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md [hunk 3]
```diff
@@ -5,254 +5,278 @@ hideTOC: true
...
-  * **Never** hide critical information in a popover. Don’t put any information into a popover that a user will always need to complete their task.
 
-**Helper text**
+#### Popovers
+Popovers are content boxes that are used for fields that might require additional background or explanation. You can also use a popover to link to external help pages or other related information. In forms, popovers are indicated by an unfilled question mark circle that a user can click on to reveal the information. They should be placed to the right of a form label. 
 
-Helper text is text below a form field that helps a user provide the right information, like _Enter a unique name_. Use helper text for information that a user needs to know in order to enter the correct information. For context or background information that is “nice to have,” use an infotip.
 
-![Helper text](./img/helper-text.png)
+ <img src="./img/popover.png" alt="Example of popover content box used in a form field" width="411">
```
Comment (redallen) [inline packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md:166]: Needs a closing `/`. Like ``

File: packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md [hunk 4]
```diff
@@ -5,254 +5,278 @@ hideTOC: true
...
+Validation on submission is performed when the user attempts to submit a form. In this scenario, the user completes the form in its entirety and the page is reloaded with any errors detected. 
 
-  * Try to limit the use of placeholder text to unfamiliar input types or specific syntax requirements (like a specific date format, number format, etc).
-  * Always precede placeholder text with "e.g." followed by a space. For example "e.g. (781)555-5555" or "e.g. 491"
-  * Keep hints concise; never extend the length of the text field.
-  * Limit the use of syntax hints to avoid overwhelming the user.
+For validation on submission, use an [inline alert](design-guidelines/usage-and-behavior/alerts-and-notifications#inline-alerts) along with field level errors wherever possible to make it simple for the user to quickly identify areas that need to be fixed.
+Notifications should describe the error and help users solve the problem.
 
+<img src="./img/error-validation.png" alt="Example of errors after validation on submission, including an inline alert at the top, and field level errors" width="459">
```
Comment (redallen) [inline packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md:203]: Needs a closing `/`. Like ``

File: packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md [hunk 5]
```diff
@@ -5,254 +5,278 @@ hideTOC: true
...
-### Enabling and disabling buttons
-* For short forms like wizards, disable submit buttons until all conditions required to complete the form are met - this reduces the need to return validation errors due to incomplete submissions.
-* For long scrolling forms, do not disable the submit button - the missing field and submit button might not be visible on the screen at the same time.
-* Once a user has submitted a form, disable the submit button to avoid duplicate submission.  
-* Use feedback messages and progress indicators like spinners or progress bars if the time to process the form might exceed users’ expectations.
+Additionally, error validation for a form in a wizard’s page should occur when or by the time the user clicks the “Next” button. You should not wait until the user has completed the whole wizard before telling them where possible errors lie.
 
-### Content
-Think of a form as a conversation between your application and your user. Apply the following principles to organize and craft content for your forms:
+<img src="./img/wizard-form.png" alt="Example of form in an in-page wizard" width="1319">
```
Comment (redallen) [inline packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md:252]: Needs a closing `/`. Like ``

File: packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md [hunk 6]
```diff
@@ -5,254 +5,278 @@ hideTOC: true
...
-**Focus on the task and context**
 
-Make sure you are clear about what the user is trying to accomplish with this form and how your users are thinking about the task they’re trying to complete.
+### Forms in a modal
+Forms in a modal should be horizontally centered with the modal background and fill up the modal width, leaving 24px between the form contents, and the modal background edges.
 
-* What are they trying to achieve?  
-* What kinds of questions will they have?
-* Where might they get stuck?
+<img src="./img/modal-form.png" alt="Example of form in a modal" width="384">
```
Comment (redallen) [inline packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md:258]: Needs a closing `/`. Like ``

File: packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md [hunk 7]
```diff
@@ -5,254 +5,278 @@ hideTOC: true
...
+Multi-column forms can be useful when you have limited vertical space. However, as a general rule, PatternFly recommends only using multi-column forms if you have extra space and avoiding them otherwise. They can be more confusing to the user and increase the chances of a user missing out on a form field.
+
+If you do decide to group certain columns together, they should be organized so that field information on the same line is related to one another. Common cases for this include:
+* First name, Middle initial, Last name
+* City, State, Zip code
+* Credit card number, Expiration date, Security code
 
-* **Include the most critical information in the field label.**
-  Answer the most basic question first: what data do I need to provide here? The field label you choose will depend on the context of your form. For example, is it enough to request a name, or do you need to specify first name, last name, and middle initial?
+<img src="./img/multi-column-eg.png" alt="Example of correct and incorrect multi-column forms" width="1275">
```
Comment (redallen) [inline packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md:271]: Needs a closing `/`. Like ``

File: packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md [hunk 8]
```diff
@@ -5,254 +5,278 @@ hideTOC: true
...
-* **Include critical secondary information as helper text.**
-  Critical secondary information is information a user must have in order to provide the right input. For example, a user creating a name for a new asset may need to specify a unique name for that asset. Letting the user know this up front helps avoid unnecessary errors.
 
-* **Include background information/context as field-level help in an popover.**
-  For example, you could use this type of help to define an unfamiliar term or concept that a novice user might not be familiar with. **Never hide critical information in an popover.**
+You should never have multi-column forms that read like a book, where you finish the left side of the form first, then move on to the right side. The form itself should still be filled from top to button and should not have two unrelated fields on the same line.
 
-* **Use natural language**
-  Again, think of a form as a conversation with your user. No one wants to have a conversation with a robot.
+<img src="./img/multi-column-read.png" alt="Visual of how multi-column forms should be read and presented" width="1275">
```
Comment (redallen) [inline packages/v4/src/content/design-guidelines/usage-and-behavior/forms/forms.md:276]: Needs a closing `/`. Like ``

**Pull request merged**